### ISSUE TITLE
random crashes

### ISSUE THREAD

**Issue by GoogleCodeExporter** [KGIII/pino-twitter#250]
```
What steps will reproduce the problem?
1. clicking the tray icon, happens nearly always. im not using the message 
indicator applet  thing

ubuntu 10.04 64bit

Assertion 'pthread_setspecific(t->key, userdata) == 0' failed at 
pulsecore/thread-posix.c:196, 
function pa_tls_set(). Aborting.
Aborted

```

Original issue reported on code.google.com by `exle...@gmail.com` on 8 May 2010 at 2:14

**Comment by GoogleCodeExporter** [comment]
```
Seems to be a duplicate of 
http://code.google.com/p/pino-twitter/issues/detail?id=245

Same issue in Fedora 12.
```

Original comment by `darksh...@gmail.com` on 8 May 2010 at 4:37

**Comment by GoogleCodeExporter** [comment]
```
I have done some research. Seems to be related to a bug in pulseaudio and some 
kind
of memory corruption. Reading similar issues on other projects, pa_tls_set() 
function
should not fail unless the key is invalid.

Using gstreamer-properties and setting ALSA as default output plugin, *seems* 
*to*
*solve* *the* *issue*.

```

Original comment by `darksh...@gmail.com` on 11 May 2010 at 10:00

**Comment by GoogleCodeExporter** [comment]
```
I have found this bug too late (already opened #265)

As far as I could find, this is linked to bad usage of libxml2's 
xmlCleanupParser().
This is documented at
http://0pointer.de/blog/projects/beware-of-xmlCleanupParser and
http://0pointer.de/blog/projects/beware-of-rsvg-term.html

It seems to be used at several places in Pino's code:
$ rgrep -i cleanup *
src/rest_api_abstract.vala:     Parser.cleanup();
src/rest_api_timeline.vala:     Parser.cleanup();
src/rest_api_timeline.vala:     Parser.cleanup();
src/rest_api_direct.vala:       Parser.cleanup();
src/rest_api_re.vala:       Parser.cleanup();
src/rest_api_acc.vala:      Parser.cleanup();

According to some comments in the posts stated hereunder, this function 
shouldn't be
called in standard circumstances.

« So, please, if your project links against libxml2 verify that it calls
xmlCleanupParser() only once, and right before exiting! And if it calls it more 
often
or somewhere else, then please fix that! »

Cheers,
Julien
```

Original comment by `Valro...@gmail.com` on 18 May 2010 at 4:52
